src/mapP.js

```diff
@@ -1,8 +1,30 @@
 const curry = require('ramda/src/curry')
 const map   = require('ramda/src/map')
 
-// mapP :: Functor f => (a -> Promise b) -> f a -> Promise f b
-const mapP = (f, list) =>
+const mapArrayP = (f, list) =>
   Promise.all(map(f, list))
 
+const mapObjectP = (f, list) => {
+  const result = {}
+  const promises = []
+
+  for (const key in list) {
+    const p = f(list[key]).then(val => { result[key] = val })
+    promises.push(p)
+  }
+
+  return Promise.all(promises).then(() => result)
```

Comment:
I think the `Promise.all` is good, but we can harness that and a little `R.zipObj` to simplify things and avoid some imperative looping:
```js
const mapObjP = (f, obj) => {
  const keys = R.keys(obj)
  const vals = keys.map(key => f(obj[key], key))

  return Promise.all(vals).then(R.zipObj(keys))
}
```

Reply:
The code is definitely cleaner, but there's an extra iteration that i'm a little sad about.